internal/controller/controllers/installenv_controller_test.go

```diff
@@ -151,6 +166,117 @@ var _ = Describe("installEnv reconcile", func() {
+			dnsGW             = "192.168.126.1"
+		)
+		hostStaticNetworkConfig := common.FormatStaticConfigHostYAML(
+			nicPrimary, nicSecondary, ip4Primary, ip4Secondary, dnsGW,
+			models.MacInterfaceMap{
+				&models.MacInterfaceMapItems0{MacAddress: macPrimary, LogicalNicName: nicPrimary},
+				&models.MacInterfaceMapItems0{MacAddress: macSecondary, LogicalNicName: nicSecondary},
+			})
+		nmstateConfig := newNMStateConfig("NMStateConfig", testNamespace, NMStateLabelValue,
+			adiiov1alpha1.NMStateConfigSpec{
+				MacInterfaceMap: []*adiiov1alpha1.MacInterfaceMapItems{
+					{LogicalNicName: nicPrimary, MacAddress: macPrimary},
+					{LogicalNicName: nicSecondary, MacAddress: macSecondary},
+				},
+				NetworkYaml: hostStaticNetworkConfig.NetworkYaml,
+			})
+		Expect(c.Create(ctx, nmstateConfig)).To(BeNil())
+		clusterDeployment := newClusterDeployment("clusterDeployment", testNamespace, getDefaultClusterDeploymentSpec("clusterDeployment-test", "pull-secret"))
+		Expect(c.Create(ctx, clusterDeployment)).To(BeNil())
+		mockInstallerInternal.EXPECT().GetClusterByKubeKey(gomock.Any()).Return(backEndCluster, nil)
+		mockInstallerInternal.EXPECT().GenerateClusterISOInternal(gomock.Any(), gomock.Any()).
+			Do(func(ctx context.Context, params installer.GenerateClusterISOParams) {
+				Expect(params.ClusterID).To(Equal(*backEndCluster.ID))
+				Expect(params.ImageCreateParams.ImageType).To(Equal(models.ImageTypeMinimalIso))
+				Expect(params.ImageCreateParams.StaticNetworkConfig).To(Equal([]*models.HostStaticNetworkConfig{hostStaticNetworkConfig}))
+
+			}).Return(&common.Cluster{Cluster: models.Cluster{ImageInfo: &models.ImageInfo{DownloadURL: "downloadurl"}}}, nil).Times(1)
+		installEnvImage := newInstallEnvImage("installEnvImage", testNamespace, adiiov1alpha1.InstallEnvSpec{
+			NMStateConfigLabelSelector: metav1.LabelSelector{MatchLabels: map[string]string{SelectorNMStateConfigNameLabel: NMStateLabelValue}},
+			ClusterRef:                 &adiiov1alpha1.ClusterReference{Name: "clusterDeployment", Namespace: testNamespace},
+		})
+		Expect(c.Create(ctx, installEnvImage)).To(BeNil())
+		res, err := ir.Reconcile(newInstallEnvRequest(installEnvImage))
+		Expect(err).To(BeNil())
+		Expect(res).To(Equal(ctrl.Result{}))
+
+		key := types.NamespacedName{
+			Namespace: testNamespace,
+			Name:      "installEnvImage",
+		}
+		Expect(c.Get(ctx, key, installEnvImage)).To(BeNil())
+		Expect(conditionsv1.FindStatusCondition(installEnvImage.Status.Conditions, adiiov1alpha1.ImageCreatedCondition).Message).To(Equal(adiiov1alpha1.ImageStateCreated))
+		Expect(conditionsv1.FindStatusCondition(installEnvImage.Status.Conditions, adiiov1alpha1.ImageCreatedCondition).Reason).To(Equal(adiiov1alpha1.ImageCreatedReason))
+		Expect(conditionsv1.FindStatusCondition(installEnvImage.Status.Conditions, adiiov1alpha1.ImageCreatedCondition).Status).To(Equal(corev1.ConditionTrue))
+	})
+
+	It("create new installEnv image with an invalid nmstate config - fail", func() {
+		var (
```

Comment:
if you have the same definition for all the related tests then you can put them under a single context and define the variables there.

Reply:
Done